remill/Arch/Name.cpp

```diff
@@ -28,7 +28,7 @@ ArchName GetArchName(const std::string &arch_name) {
   } else if (arch_name == "x86_avx512") {
     return kArchX86_AVX512;
 
-  } else if (arch_name == "amd64") {
+  } else if (arch_name == "amd64" || arch_name == "x86_64") {
```

Comment:
This logic should go in `GetModuleArch`. Then, you can also inspect the attributes (target features) of individual functions [1] to infer the `_avx` or `_avx512` suffixes. This could mean that in anvill, you would need to get the target arch on a per-function basis, possibly with some caching on attribute sets or the like.

[1] https://github.com/lifting-bits/remill/blob/master/remill/Arch/Arch.cpp#L835-L852